File: Android/tiwari-aditi_AditiTiwari_2125ec1138_2/GettingStarted/RotateANumber.cpp
#include<iostream>
#include<math.h>
using namespace std;

int main(){
    int n,k,r,temp1,len,m1,m2;
    cin>>n;
    cin>>k;
    temp1 = n;
    
    while(n!=0){
        len++;
        n = n/10;
    }

    if(k>0){
        m1 = pow(10,k);
        m2 = pow(10,len-k);
        r = temp1 % m1;
        r = r*m2;
        temp1 = temp1/m1;
        r = r+temp1;
    }
    else if(k<0){
        k = -k;
        m1 = pow(10,k);
        m2 = pow(10,len-k);
        r = temp1 % m2;
        r = r *m1;
        temp1 = temp1/m2;
        r = r+temp1;
    }
    
    cout<<r;
    return 0;
}

File: Android/sukritoberoi_SukritKaurOberoi_2125cse1065_2/GettingStarted/RotateANumber.java
import java.util.*;
   
   public class Main{
   
   public static void main(String[] args) {
     // write your code here  
     long n,l,n1;
     long k,c=0,i;
     Scanner sc=new Scanner(System.in);
     n=sc.nextLong();
     k=sc.nextInt();
     n1=n;
     while(n1!=0)
         {
             c++;
             n1=n1/10;
         }
     if(k%c==0)
      System.out.println(n);
     else if(k>0)
     {
         for(i=0;i<k;i++)
         {
           l=n%10;
           n=n/10;
           n=(Math.round(Math.pow(10,c-1))*l)+n;
         }
         System.out.println(n);
     }
     else
     {
         for(i=k;i>0;i--)
         {
             l=n%Math.round(Math.pow(10,c-1));
             n=n/Math.round(Math.pow(10,c-1));
             n=n*10+l;
           
         }
         System.out.println(n);
     }
    }
   }